message_definitions/v1.0/development.xml

```diff
@@ -692,22 +730,67 @@
-      <field type="uint32_t" name="error_flags" enum="GENERIC_PAYLOAD_ERROR_FLAGS" display="bitmask">Errors</field>
-      <field type="uint32_t" name="custom_error_flags" display="bitmask">Bitmap used to show custom error flags.</field>
-      <field type="float" name="power_draw" units="W" invalid="NaN">The power draw of the generic payload. NaN: field not provided</field>
-      <field type="uint16_t" name="weight" units="g" invalid="0">Generic payload weight. 0: field not provided.</field>
-      <field type="float" name="temp_c" invalid="NaN">Temperature in Celsius. NaN: field not provided.</field>
+      <description>Status and discovery message for a generic payload. This should be broadcast at a low rate (nominally 1 Hz) to announce the payload's presence and capabilities.</description>
+      <field type="uint32_t" name="uptime_ms" units="ms">Time since payload boot.</field>
+      <field type="uint32_t" name="error_flags" enum="GENERIC_PAYLOAD_ERROR_FLAGS" display="bitmask">Current error flags.</field>
+      <field type="uint32_t" name="custom_error_flags" display="bitmask">Bitmap for custom error flags specific to this payload type.</field>
+      <field type="uint16_t" name="num_functions">Number of functions this payload provides.</field>
+      <field type="uint16_t" name="num_telemetry_channels">Number of telemetry channels this payload provides.</field>
+      <field type="char[32]" name="name">Name of the generic payload for UI display. NULL terminated string.</field>
+      <extensions/>
+      <field type="uint16_t" name="power_draw" units="mW" invalid="0">Power consumption of the payload. 0 if not available.</field>
+      <field type="uint16_t" name="temperature" invalid="UINT16_MAX">Temperature of the payload (0.002 degrees Celsius per unit, 0 = -40C). UINT16_MAX if not available.</field>
+      <field type="uint16_t" name="weight" units="g" invalid="0">Weight of the payload. 0 if not available.</field>
     </message>
     <message id="399" name="GENERIC_PAYLOAD_FUNCTION_STATUS">
       <description>Generic payload function status.</description>
-      <field type="char[32]" name="name">Generic payload function name to be used in UI. This is a NULL terminated string. If it is exactly 32 characters long, add a terminating NULL. If this string is empty, a generic name is shown to the user.</field>
       <field type="uint16_t" name="index">Index of this function on the generic payload.</field>
+      <field type="uint8_t" name="type" enum="GENERIC_PAYLOAD_FUNCTION_TYPE">Type of function.</field>
+      <field type="uint8_t" name="value_type" enum="GENERIC_PAYLOAD_VALUE_TYPE">Data type for value/min/max fields.</field>
+      <field type="uint8_t" name="enabled">0: Disabled, 1: Enabled</field>
+      <field type="uint8_t[4]" name="value_low">Lower 32 bits of current value. For 32-bit types, this is the entire value. For 64-bit types, this is the least significant word.</field>
```

Comment:
there is often confusion with MAV_CMDs that have a data type of float but are supposed to be interpreted as bool/int through casts, i think i having a generic "union" type works better